include/commissioner/commissioner.hpp

```diff
@@ -1147,6 +1161,36 @@ class Commissioner
      *         git repository.
      */
     static std::string GetVersion(void);
+
+    /**
+     * @brief Asynchronously query diagnostic decoded data from a Thread device.
+     *
+     * This method sends a DIAG_GET.qry message to the specified Thread device,
+     * requesting the set of diagnostic data indicated by `aDiagDataFlags`.
+     * The response, or any errors encountered, will be delivered to the provided `aHandler`.
```

Comment:
Could you explain how the requested Diag data will be received? I know it's through the `CommissionerHandler` callbacks, but it should be documented here.